src/main/scala/chisel3/util/SRAM.scala

```diff
@@ -78,6 +81,45 @@ class MemoryReadWritePort[T <: Data](tpe: T, addrWidth: Int, masked: Boolean) ex
   }
 }
 
+@instantiable
+/** Object Module for SRAM, used for storing metadata of an SRAM. */
+class SRAMOM extends Class {
+  val depth:           Property[Int] = IO(Output(Property[Int]()))
```

Comment:
The size of a memory is a BigInt, not an Int (mainly for use in simulation but still) so I think this should probably be a BigInt, do we support `Property[BigInt]` or does that matter @mikeurbach?

Reply:
We do support Property[BigInt]. It's all the same (arbitrary precision signed integers) in FIRRTL, but we should probably use BigInt here if the size of the memory is BigInt.